src/plots/cartesian/axes.js

```diff
@@ -927,6 +959,14 @@ axes.calcTicks = function calcTicks(ax, opts) {
             _minor // noSuffixPrefix
         );
 
+        if(tickVals[i].skipLabel) {
+            t.skipLabel = _skipLabel;
+
+            if(!firstLabelCreated) ax._prevDateHead = '';
```

Comment:
This ensures we get the second line on the first tick, whatever that is... but not if the second line changes in the middle of the axis and the first tick with the new second line is removed. This tends to happen with multi-day but sub-month ticks, but could happen in other cases if you set an unusual `tick0`.

showing all labels:
<img width="488" alt="Screen Shot 2022-02-04 at 09 16 35" src="https://user-images.githubusercontent.com/2678795/152545435-709470fc-d63c-4d36-9f5e-b04a75dffc5e.png">
`ticklabelstep=2`, notice there's no 2610 but there should be.
<img width="482" alt="Screen Shot 2022-02-04 at 09 16 49" src="https://user-images.githubusercontent.com/2678795/152545433-573f6ff4-f1a3-4b39-a387-8e9b9e7686dc.png">

Reply:
Good catch!
See new commits under https://github.com/plotly/plotly.js/pull/6088/commits/40a55328778f697798f3ab14c8ff6d95a76f5e0b.